pkg/operator/controllers/dnsmasq/dnsmasq.go

```diff
@@ -110,6 +110,46 @@ chmod 0744 /etc/resolv.conf.dnsmasq
 /bin/rm $TMPNETRESOLV
 /bin/rm $TMPSELFRESOLV
 {{ end }}
+
+{{ define "99-dnsmasq-restart" }}
+#!/bin/sh -x
+# This is a NetworkManager dispatcher script to restart dnsmasq
+# in the event of a network interface change (e. g. host servicing event https://learn.microsoft.com/en-us/azure/developer/intro/hosting-apps-on-azure)
+# this will restart dnsmasq, reapplying our /etc/resolv.conf file and overwriting any modifications made by NetworkManager
+
+interface=$1
+action=$2
+
+log() {
+    logger -i "$0" -t '99-DNSMASQ-RESTART SCRIPT' "$@"
+}
+
+if [[ $interface == eth* && $action == "up" ]] || [[ $interface == eth* && $action == "down" ]]; then
```

Comment:
In the test, the interface that gets unplugged / replugged (the VF interface) is called enP53531s1. Maybe we look for interfaces called `e*` or specifically eth* plus enP*?